pkg/lib/dynatrace/dashboards.go

```diff
@@ -0,0 +1,62 @@
+package dynatrace
+
+import (
+	"github.com/keptn-contrib/dynatrace-service/pkg/common"
+	log "github.com/sirupsen/logrus"
+	"strings"
+)
+
+// DynatraceDashboards is the data structure for /dashboards endpoint
+type DynatraceDashboards struct {
+	Dashboards []struct {
+		ID    string `json:"id"`
+		Name  string `json:"name"`
+		Owner string `json:"owner"`
+	} `json:"dashboards"`
+}
+
+// SearchForDashboardMatching searches for a dashboard that matches either project, service or stage
```

Comment:
No error was introduced due to the refactoring, but actually, we would like to have an exact match of project, service, and stage. Right now, only either project, service or stage needs to match.

Reply:
I have added several tests to verify that only full matches of `project`, `service` and `stage` are allowed. Which is the actual behavior - so I just updated the documentation of the method.

Please have a look at: https://github.com/keptn-contrib/dynatrace-service/pull/415/commits/d5c398a9bc6511d10875a51c3c9d071ade8688f1

Reply:
added further tests to also verify that no subsets of `project`, `service` and/or `stage` are matched. Please see: https://github.com/keptn-contrib/dynatrace-service/pull/415/commits/1c0663b1a146f28557350f5783f308056fe9a1e4

Reply:
Thank you! 
My assumption was wrong and the tests verified it.